aten/src/ATen/native/ScatterGatherChecks.h

```diff
@@ -34,12 +34,17 @@ static void scatter_gather_dtype_check(
 // Test:
 // 1. index.size(d) == self.size(d) for all d != dim
```

Comment:
isn't this true for all d?  You resize this stuff anyway so I don't know what the check for dim below does, but seems like we should be consistent?

Reply:
I totally agree, yes. Shall we fix it here or in a separate PR?